src/resource/filters.js

```diff
@@ -194,15 +198,38 @@ export default function filtersFactory($container, config) {
             this.render($container);
         })
         .on('render', function() {
-            if (this.config.data) {
-                this.update(this.config.data);
-            }
+            Promise.race([
+                new Promise(resolve => {
+                    if (this.config.data) {
+                        this
+                            .on('update.ready', () => {
+                                this.off('update.ready');
+                                resolve();
+                            })
+                            .update(this.config.data);
+                    } else {
+                        resolve();
+                    }
+                }),
+                new Promise((resolve, reject) => {
```

Comment:
As a nice to have, what about extracting this pattern to something like `core/promiseTimeout`

Reply:
Agree, here is the PR:
https://github.com/oat-sa/tao-core-sdk-fe/pull/21